src/OrchardCore.Modules/OrchardCore.Localization/Startup.cs

```diff
@@ -73,4 +77,41 @@ public override void ConfigureServices(IServiceCollection services)
             services.Configure<RequestLocalizationOptions>(options => options.ApplyCurrentCultureToResponseHeaders = true);
         }
     }
+
+    [Feature("OrchardCore.Localization.CulturePicker.Admin")]
+    public class CulturePickerStartup : StartupBase
+    {
+        private static readonly Task<ProviderCultureResult> NullProviderCultureResult = Task.FromResult(default(ProviderCultureResult));
+
+        private readonly AdminOptions _adminOptions;
+
+        public CulturePickerStartup(IOptions<AdminOptions> adminOptions)
+        {
+            _adminOptions = adminOptions.Value;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<IShapeTableProvider, AdminCulturePickerShapes>();
+
+            services.Configure<RequestLocalizationOptions>(options => options.AddInitialRequestCultureProvider(new CustomRequestCultureProvider(context =>
+            {
+                var path = context.Request.Path.Value[1..];
+                if (path.Equals(_adminOptions.AdminUrlPrefix, StringComparison.OrdinalIgnoreCase) ||
+                    path.StartsWith(_adminOptions.AdminUrlPrefix + "/", StringComparison.OrdinalIgnoreCase))
```

Comment:
When you inject `_adminOptions` you could compute once _adminOptions.AdminUrlPrefix + "/" and store it in a field.

Reply:
Why not ;)